src/util/createInstantSearch.js

```diff
@@ -0,0 +1,108 @@
+import instantsearch from 'instantsearch.js/es';
+import algoliaHelper from 'algoliasearch-helper';
+const { SearchParameters, SearchResults } = algoliaHelper;
+import { _objectSpread } from './polyfills';
+import { warn } from './warn';
+
+export const createInstantSearch = instantSearchOptions => {
+  const search = instantsearch(instantSearchOptions);
+  const { searchClient, indexName } = instantSearchOptions;
+
+  // main API for SSR, called in asyncData of a root component which contains instantsearch
+  search.findResultsState = params => {
```

Comment:
Already discussed but `getResultsState` would make more sense IMO (although it's named `findResultsState` in React InstantSearch for now).